### PR TITLE
Change to REST API

### PR DESCRIPTION
**Pull request by itan1**
I refactored the `MLFLogger` to use the REST API from [MLFlowClient.jl](https://github.com/JuliaAI/MLFlowClient.jl), see issue #3.

I could keep mostly the same interface with only some minor differences.

### CODE REVIEW

**Comment by rejuvyesh** [comment]
Fantastic. Just one quick thing. The python based mlflow clients usually also handle a lot of cloud authentication stuff (for example logging in azureml is easier if you have [`azureml-mlflow` installed together with `mlflow`](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-log-view-metrics?view=azureml-api-2&tabs=interactive). I'm guessing, we will have to lose on those things if we go the Julia only route until someone implements that part of the API.

**Comment by rejuvyesh** [comment]
Do you wish to take a shot at enabling the server at MLFLOW_URI in the github workflow to allow running the tests. We probably just need to install and run [`mlflow serve`](https://mlflow.org/docs/latest/tracking/server.html#start-the-tracking-server) and set the environment variable.

**Comment by itan1** [comment]
I indeed don't see anything azure specific in MLFlowClient.jl at the moment. Could we make a separate issue for that functionality?

Using the [MLFlow basic authentication](https://mlflow.org/docs/latest/auth/index.html) should be possible via the `headers` argument like described in [this MLFlowClient Issue](https://github.com/JuliaAI/MLFlowClient.jl/issues/33#issuecomment-1830541659). The `MLFLogger` constructor should pass the argument to the `MLFlow` constructor.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -23,12 +23,15 @@ jobs:
...
+      - name: Setup python and mlflow server
         uses: actions/setup-python@v1
         with:
           python-version: ${{ matrix.python-version }}
           architecture: ${{ matrix.arch }}
-      - run: python -m pip install setuptools mlflow  
+      - run: |
+          python -m pip install mlflow
+          python -m pip show mlflow
+          mlflow server --host localhost --port 5000
```
Comment (rejuvyesh) [inline .github/workflows/CI.yml]:
```suggestion
          mlflow server --host localhost --port 5000 &
```

**Comment by itan1** [comment]
Thanks @rejuvyesh! Could we maybe tag a new version for this?

**Pull request merged**